### ISSUE TITLE
0.55.x Discussion

### ISSUE THREAD

**Issue by grabbou** [react-native-community/react-native-releases#11]
Please report any major issues and commits to cherry-pick in this thread for discussion.

**Comment by gengjiawen** [comment]
The android ci is still broken in master branch, which stuck my pr.Can you push to this fixed ? I tried but I know few about buck.

**Comment by turnrye** [comment]
Hey @gengjiawen, we're aware of the issue with CI and a fix is in progress.

**Comment by gengjiawen** [comment]
@turnrye It is still broken, I don't think it's a small problem because it stuck all community pr getting merged. I really wish we can fix it asap.

**Comment by grabbou** [comment]
Yeah, I am going to spend some time this week making it green because I need to cut next release branch.

Meanwhile, yet another cherry-pick request to consider: https://github.com/facebook/react-native/issues/18223#issuecomment-376651552

**Comment by dannycochran** [comment]
@grabbou 

Fixes an issue with invertStickyHeaders not being passed to ScrollView from SectionList / FlatList:

https://github.com/facebook/react-native/pull/18524

**Comment by gengjiawen** [comment]
Hopefully the ci thing will be fixed in next week :(

**Comment by grabbou** [comment]
All prior commits have been cherry-picked and we are releasing stable version now.

**Comment by kelset** [comment]
Hey @grabbou looks like there's a Metro issue: https://github.com/facebook/react-native/issues/18681#issuecomment-378441952

**Comment by grabbou** [comment]
Yeah, seems to be the case. I'll look into it before making a new one.

**Comment by grabbou** [comment]
Yeah, someone messed up `runIOS` quite a bit in this release. We don't test this command during release. I've manually tested both issues (reproduced) and fixed inline inside 0.55 branch. Should be ready to go.

**Comment by Gyran** [comment]
Would be nice if https://github.com/facebook/react-native/commit/c787e0e25216d04a76ce2e3c0c88218ab9db3545 could be cherry-picked so that we can build using a Podspec environment.

**Comment by kelset** [comment]
BTW looks like there's an issue with WebSockets: https://github.com/facebook/react-native/issues/18696

Which I'm not sure is a 'wanted' side effect (meaning, now it's always need to pass down 3 arguments) or a real bug.

**Comment by grabbou** [comment]
I'll be cherry-picking this one:
https://github.com/facebook/react-native/pull/18721 to stop issues with
`run-ios` once forever.

**Comment by sjmueller** [comment]
It looks like this commit got included in `0.55`:
https://github.com/facebook/react-native/commit/16c9e5b

However shortly afterwards it got reverted due to a bug:
https://github.com/facebook/react-native/commit/8466db0fd355aea778ff71cfb10c927cfd18a551

In our case there are other bugs also associated with this commit (we've discovered a problem with `FlatList.scrollToIndex` that also broke), so the reversion is pretty important.

**Comment by kelset** [comment]
Yet another `run-ios` related PR: https://github.com/facebook/react-native/pull/18710#issuecomment-379282644

**Comment by grabbou** [comment]
@kelset not really something we need to address right now, but I'll keep track of it.

**Comment by itchingpixels** [comment]
This would resolve a long outstanding issue where Yoga (and the whole of react-native) can not be compiled when used with Cocoapods https://github.com/facebook/react-native/pull/18492

**Comment by Leeds-eBooks** [comment]
Hi, really sorry if it's not my place to comment here, or if this isn't the right place for this kind of comment, but there are a lot of people waiting for this PR to be merged and released: https://github.com/facebook/react-native/pull/18278, it fixes a major bug with `TextInput` on iOS.

How can we register our desire to get this released?

Thanks for your help!

**Comment by ptomasroos** [comment]
Would be great to have the latest fix for the Podspec issue to 0.55.2 unusable otherwise. https://github.com/facebook/react-native/commit/c787e0e25216d04a76ce2e3c0c88218ab9db3545

**Comment by rozele** [comment]
This one will also have to be cherry-picked (WebSocketModule regression on Android): https://github.com/facebook/react-native/commit/4b6e9d3dfd213689c5b319a4efee8d4f606d6f9d

**Comment by alax** [comment]
iOS controlled TextInputs are still totally broken as of 55.2, we still need https://github.com/facebook/react-native/pull/18278 merged (or a similar fix)

**Comment by turnrye** [comment]
@charpeni asked for this to be cherry picked: https://github.com/facebook/react-native/commit/c787e0e25216d04a76ce2e3c0c88218ab9db3545

**Comment by grabbou** [comment]
@alax I've pinged the reviewers to merge it. Before that happens, please keep this discussion exclusive to commits that landed into master already.

**Comment by grabbou** [comment]
Releasing 0.55.3 with the fixes now.

**Comment by grabbou** [comment]
Someone requested this: https://github.com/facebook/react-native/commit/f5207ba9c764f33ef83fa897f6014d67193be0e2 in different issue

**Comment by grabbou** [comment]
Another request: https://github.com/facebook/react-native/commit/9d45de60accda028d11b715a3407119353a81d2c#diff-82c5d95c19a17c3f2375f0b1e09f6da4

**Comment by rodan-lewarx** [comment]
What about https://github.com/facebook/react-native/commit/bfcfe7961db0970e2575eafe2f3c9c668bd8940d ? Would it be possible to get it to 0.55?

**Comment by jkimbo** [comment]
We are seeing this error a lot in our app: https://github.com/facebook/react-native/issues/18709 . https://github.com/facebook/react-native/commit/7c5d581d7848206603cf9d80a841afb49759ec1a seems to fix it. Any chance of it being included in another patch release?

**Comment by kelset** [comment]
Prob we'll do a 'beefy' 0.55.4 soon-ish, we'll try to include those in.

**Comment by grabbou** [comment]
Yup, that's right. I will be also doing a hot-fix to unbreak `react-native
init` babel preset by pinning it to a specific version later today.

**Comment by jkimbo** [comment]
Any update on this? Is there still going to be a 0.55.4 release?

**Comment by itchingpixels** [comment]
we're eagerly waiting :)

**Comment by grabbou** [comment]
FYI there is no need for a hot-fix since we fixed babel preset the other way. I am going to cherry-pick the aforementioned commits and going to release it right now.

I will be cutting next release around the weekend, definitely before going on holidays. Please stay tuned.

**Comment by jkimbo** [comment]
@grabbou do you mean there is not going to be a 0.55.4? We are still getting a lot of errors fhat would be fixed by facebook/react-native@7c5d581 . A patch release would really help us out.

**Comment by kelset** [comment]
@jkimbo There will **probably** be a 0.55.4, we are not sure yet because the Babel 7 commit forced us to rediscuss the release plan for 0.55.x - 0.56.x - 0.57.x and since we still have issues on the CI there is no estimate on when the new releases will be out.

Rest assured that we read the comments in the repos, but given that React Native is **not** as OSS-first project there are limitations in how we are able to achieve release status and atm the internal team at FB is under heavy loads of work, so if you don't want to fork the main repo & cherry pick that commit for your team you will have to wait a bit more.

**Comment by jkimbo** [comment]
@kelset ok fair enough and thanks for the update. Hope I didn't mean to come across as ungrateful, I really appreciate the work you all put into the project, it's just useful to know whats going on so that I can work around if necessary 🙂

**Comment by grabbou** [comment]
Unfortunately, recent changes broke the CI and it is impossible to release
an update right now. The 0.55.4 version is already scheduled, sent to
Circle waiting for the tests to pass. Once the fix happens, we will be able
to push it to npm.

Unfortunately, the Facebook team members in charge of this are currently on
a leave and I can't give you any estimate when this gets unblocked. We are
in touch with the team and will do our best to make it as fast as possible.

**Comment by peacechen** [comment]
Will #18223 be considered for a 0.55 release?  This commit fixes the empty blob error "Invalid response for blob":

https://github.com/xiamx/react-native/commit/b35071b5daabf370e7789e00e1593788fcf5aecf

**Comment by grabbou** [comment]
I'll add it in.

Yes, the release has some issues on the Circleci. I need to investigate that with Facebook team. Due to holidays season and couple of people from Facebook being busy, this can happen around next week.

**Comment by sercanov** [comment]
Any development on 0.55.4 release? https://github.com/facebook/react-native/issues/18709 is really nasty

**Comment by grabbou** [comment]
I'll check that tomorrow.

Also, an update for everyone interested: I'll be working on fixing the issues and pushing the release out there. 

Apologies for the delay if you have been blocked by any of the aforementioned issues.

**Comment by grabbou** [comment]
Got request for the two following commits: 3e534b9 and e708010 that fix `isMounted()` warnings. Since this is still the latest stable that I hope many of you will use, I am going to patch it.

**Comment by grabbou** [comment]
Unfortunately, cherry-picking those is going to be quite tricky since there were many commits in the meantime. My attempt ended up in resolving different conflicts here and there that make me worried it's a bit dangerous process.

For now I'll postpone it and focus on getting 0.56 out there.

CC: @gaearon

**Comment by itay9001** [comment]
So until 0.56 we won't see a fix for the Module.close issue? Will is happen only in development or on a production release aswell?

Anyway, 0.55.4, Issue still occur.

**Comment by kpink224** [comment]
There is a huge performance issue with android TextInputs that was introduced in 0.50 in [this commit](https://github.com/facebook/react-native/commit/c550f27a4e82402b966567d7b79a8b1f1d1491a5). Fortunately, the fix has been merged into master [HERE](https://github.com/facebook/react-native/commit/c6b4f9f2ce59bc757d9e211f46294faa03df55c6).

Is there any way we can get this in a patch for 0.55? Since 0.56 will be a big upgrade, we don't want to take the risk of too many things changing since were are releasing our product soon.

**Comment by hramos** [comment]
Now that `0.56.0-rc` has been cut, I think it's time to close this issue.

@kpink224 you can try building a release based on 0.55.4 plus https://github.com/facebook/react-native/commit/c6b4f9f2ce59bc757d9e211f46294faa03df55c6 for use in your own product. Releasing an official 0.55.5 release might be convenient for your particular case, but it would come with added burden on us maintainers that is hard to justify when the fix is already present in 0.56 and we're already working on getting ready for the 0.57 cut in just over two weeks.

**Comment by grabbou** [comment]
The idea behind discussion issues has been always to provide support for "current" stable release too. That said, especially that another issue (#27) has been opened, I am going to reopen this issue and keep available for additional comments until next stable release is cut.

This is the commit that has been requested to be cherry-picked: https://github.com/facebook/react-native/commit/41975f75d96ef4b606b4618461bf24d5db063b77

**Comment by grabbou** [comment]
Regarding ^, I am aware of another person who had to fork React Native in time for a release to get this issue fixed. That said, given the minimal scope of the requested commit, I don't see any risks when it comes to backporting this change.

If there is no objection, I'll schedule next update tomorrow morning, along other commits that are pending for 0.56.0-rc.

**Comment by asowayan** [comment]
@grabbou If there is going to be another 0.55.x release, can you please cherry-pick https://github.com/facebook/react-native/commit/c6b4f9f2ce59bc757d9e211f46294faa03df55c6 ? It fixes a huge performance problem on Android.

**Comment by hramos** [comment]
@asowayan is forking React Native to get 0.55 + this commit too much trouble? Just curious - I want to understand why this approach is not common.

**Comment by davidlongest** [comment]
@hramos The actual process of forking React Native and cherry picking a commit isn't too bad. However, having the infrastructure to keep that custom build of React Native is not something that everybody has set up. Sure, we could also commit the build output to our project repo or add it as a git submodule, but that creates additional complexity and clutters up a project's git repo.

The React Native project already has a place to publish builds to and sharing some of these major performance / bug fixes with everybody that uses React Native as a point release would be very helpful.

**Comment by hramos** [comment]
I am not sure I get how big of a burden this is, we are not talking about the case when React Native is forked to add functionality that is unlikely to be considered and merged into the main repo. The commit @asowayan is referring to is already part of the next release, `0.56`, which is due to hit stable in six days.

**Comment by grabbou** [comment]
0.56.0 has been promoted to stable right now.
